sycl/source/detail/event_impl.hpp

```diff
@@ -75,7 +75,11 @@ class event_impl {
   /// Self is needed in order to pass shared_ptr to Scheduler.
   ///
   /// \param Self is a pointer to this event.
-  void wait(std::shared_ptr<sycl::detail::event_impl> Self);
+  /// \param Success is an optional parameter that indicates that the wait
```

Comment:
This comment appears in several places ( great! ) but I don't quite understand it.

So if nullptr, then it's not being used.
If the value it points to is true, then the operation succeeded.
But what if the value it points to is false? Are you saying that it failed? Or that it "could fail", meaning it "has not yet succeeded, but we don't know the final outcome yet".  

If the interpretation is the latter, should it perhaps be an enum instead of a bool?

Reply:
The former, we always know the outcome after waiting for an event. Do you think the comment needs to be changed to make that clearer?

Reply:
Yeah, probably. The language about "could fail" and "iff" lead me to think the false state wasn't failure but "possible failure".

Reply:
Done.